File: static_web/FlowFieldsGen/js/animation/events/StartEvent.js
class StartEvent extends AnimationEvent {

  constructor(actor, startingTime) {
    super();
    this.actor = actor;
    this.startingTime = startingTime;
  }

  getName() {
    return "StartEvent";
  }

  hasTimeCome(time) {
    return time > this.startingTime;
  }

  getActor() {
    return this.actor;
  }
}

File: java/FlowFieldsGen/src/main/java/in/mcxiv/animation/events/StartEvent.java
package in.mcxiv.animation.events;

import in.mcxiv.animation.Actor;
import in.mcxiv.animation.Event;

public class StartEvent extends Event {

    private final Actor actor;
    private final double startingTime;

    public StartEvent(Actor actor, double startingTime) {
        this.actor = actor;
        this.startingTime = startingTime;
    }

    @Override
    public String getName() {
        return "StartEvent";
    }

    @Override
    public boolean hasTimeCome(double time) {
        return time > startingTime;
    }

    @Override
    public Actor getActor() {
        return actor;
    }
}
